pep-0458.txt

```diff
@@ -319,16 +323,27 @@ top-level roles (including its own).  The *timestamp* role references the
 latest *snapshot* and can signify when a new snapshot of the repository is
 available.  The *snapshot* role indicates the latest version of all the TUF
 metadata files (other than *timestamp*). The *targets* role lists the file
-paths of available target files together with their hashes. The file paths must
-be specified relative to a base URL. This allows the actual target files to be
-served from anywhere, as long as the base URL can be accessed by the client.
-Each top-level role will serve its responsibilities without exception.  Figure
-1 provides a table of the roles used in TUF.
+paths of available target files together with their cryptographic hashes.
+The file paths must be specified relative to a base URL. This allows the
+actual target files to be served from anywhere, as long as the base URL
+can be accessed by the client. Each top-level role will serve its
+responsibilities without exception.  Figure 1 provides a table of the
+roles used in TUF.
 
 .. image:: pep-0458-1.png
 
 Figure 1: An overview of the TUF roles.
 
+This PEP RECOMMENDS that every metadata or target file be hashed using
+both the SHA-256 and SHA-512 functions of the `SHA-2`__ family. SHA-2
+has native and well-tested Python support (allowing for verification of
+these hashes without additional, non-Python dependencies), and using
+both functions should provide sufficient protection against
+`collision attacks`__ for the foreseeable future.
```

Comment:
I feel it could be worth expanding on this a bit to briefly discuss _how_ having two hashing algorithms, and two copies of every target file, helps in the case of a collision attack.

Reply:
The assumption (justified or not) we are making here is that an attack that works for SHA-256 does not easily translate to SHA-512. Personally, I'd much rather use SHA-512 and SHA-3, because they're both very different designs. Does anyone have strong thoughts here?

Reply:
Well I'm imagining that once an attack on either is announced it'll move from technically to financially unfeasible, giving people enough time to migrate no? I agree that at the least using SHA512 (Merkle-Damgard) vs SHA3 (Sponge) makes finding a fundamental design flaw on both less likely.